scripts/new_avl.py

```diff
@@ -0,0 +1,569 @@
+##adapted from github of Pavel Grafov: https://github.com/pgrafov
```

Comment:
If this code is mostly copied from https://github.com/pgrafov/python-avl-tree we will need to properly acknowledge the contributed code.  Fortunately, it was released under the MIT license which will be compatible with our BSD license.